src/coverlet.core/Extensions/InstructionExtensions.cs

```diff
@@ -0,0 +1,247 @@
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+
+namespace Coverlet.Core.Extensions
+{
+    // Extension methods that can smooth out small differences between similar CIL
+    // instructions, such as "ldarg.0" as opposed to "ldarg 0".
+    public static class InstructionExtensions
```

Comment:
If it's not strictly necessary know the index of arg I prefer avoid helper like this for now and keep pattern recognition simple, also because arg index could change in future version of compiler, but a load is a load forever, I hope 😄 

So the question is, does this index id make difference for pattern recognition or can collide with other pattern?
Are there collision sample at the moment?

Reply:
The reason I was checking the index wasn't so much because I needed the index to be something particular like 0, but because I wanted to check that the index used in two different instructions was the same.  So my main goal was to be able to do this:

```
                else if (instructions[currentIndex - 1].IsLdloc(out int localVariableIndex) &&
                         instructions[currentIndex + 1].IsLdloc(out int reloadedLocalVariableIndex) &&
                         localVariableIndex == reloadedLocalVariableIndex &&
                         instructions[currentIndex + 2].OpCode == OpCodes.Callvirt &&
                         instructions[currentIndex + 2].Operand is MethodReference method &&
                         method.DeclaringType.FullName == "System.IAsyncDisposable" &&
                         method.Name == "DisposeAsync")
```

Of course, the pattern would be simpler if I didn't do that, but it also seemed like it was becoming simple enough that it might intersect with some other call to `DisposeAsync` that appeared in user-written code.

Still, I don't think there was an example of where this would lead us astray, so I'll try removing all of that, see if the tests pass, and, if so, will take out the helpers and move on with the simpler approach.

Reply:
My other goal was to be able to ask "Is this a local variable load?" without caring what the index was, but the only way I could figure out how to do that reliably was this:

```
                else if ((instructions[currentIndex - 1].OpCode == OpCodes.Ldloc ||
                          instructions[currentIndex - 1].OpCode == OpCodes.Ldloc_S ||
                          instructions[currentIndex - 1].OpCode == OpCodes.Ldloc_0 ||
                          instructions[currentIndex - 1].OpCode == OpCodes.Ldloc_1 ||
                          instructions[currentIndex - 1].OpCode == OpCodes.Ldloc_2 ||
                          instructions[currentIndex - 1].OpCode == OpCodes.Ldloc_3) &&
```

Because unless we know what the index is, then a load from local variable index 1 could theoretically be a `ldloc 1`, a `ldloc.s 1`, or a `ldloc.1`.  But if I don't know which index I care about, then I have to allow for the fact that it might be any of these.

Reply:
Got it, let's see....never had a collision like that in past, but you could be right.